homeassistant/components/netatmo/data_handler.py

```diff
@@ -136,6 +135,9 @@ async def async_fetch_data(self, data_class_entry):
         try:
             await self.data[data_class_entry].async_update()
 
+        except AttributeError as err:
```

Comment:
Why don't we guard against a missing data class instead?

Reply:
In the end I want to do exactly that. But since I can't replicate this issue I want to gather some insights this way. If you think this is a bad idea I'll add a guard right away instead.

Reply:
We're setting the data class to None below. Where are we handling that? It seems we should assume it can be None and guard against that before accessing it as an instance with methods.

Reply:
That's right. I've replaced the exception handling by a guard.